File: CSharp/WinForm/Src/frmLogotron.cs

using System;
using System.Collections.Generic;
//using System.ComponentModel;
//using System.Data;
//using System.Drawing;
//using System.Linq;
using System.Text;
//using System.Threading.Tasks;
using System.Windows.Forms;

using UtilFichier = UtilWinForm.FileHelpers;
using LogotronLib;
using Logotron.Src; // frmQuiz
using LogotronLib.Src;
//using System.Runtime.CompilerServices;
using Logotron.Src.Util; // clsMsgDelegue

namespace Logotron
{

    public partial class frmLogotron : Form
    {
        private bool bDebug = clsConstWinForm.bDebug;
        bool m_bMajViaCode=false;
        
        public clsMsgDelegue m_msgDelegue;

        private void MsgBox(object sender, clsMsgEventArgs e)
        {
            UtilWinForm.clsMessageUtil.MsgBox(e.sMessage);
        }

        private void MsgBox(string sTxt)
        {
            UtilWinForm.clsMessageUtil.MsgBox(sTxt);
        }

        private int m_iIndex = 0;
        private void AfficherTexte(string sTxt)
        {
            UtilWinForm.clsUtil.AfficherTexteListBox(
                sTxt, ref m_iIndex, this, this.lbResultats);
        }

        public frmLogotron()
        {
            InitializeComponent();
            if (bDebug) this.StartPosition = FormStartPosition.CenterScreen;
            
            m_msgDelegue = new clsMsgDelegue();
            EventHandler<clsMsgEventArgs> obj = this.MsgBox;
            m_msgDelegue.EvAfficherMessage += obj;
            
            // S'il y a besoin de rafraichir (pas besoin pour un MsgBox
            //  mais util pour une barre de msg par exemple)
            //EventHandler<clsDoEventsEventArgs> obj2 = this.DoEvents;
            //m_msgDelegue.EvDoEvents += obj2;
        }
        
        private void DoEvents(object sender, clsDoEventsEventArgs e)
        {
            Application.DoEvents(); 
        }

        private void Form1_Load(object sender, EventArgs e)
        {
            Initialisations();
        }

        private void Initialisations()
        {
            string sTxt = Text + " - V" + clsConstWinForm.sVersionAppli +
                " (" + clsConstWinForm.sDateAppli + ")";
            if (bDebug) sTxt += " - Debug";
            this.Text = sTxt;

            Util.clsAppUtil.AppTitle = clsConstWinForm.sNomAppli; 

            m_bMajViaCode = true;
            this.lbNiveau.SetSelected(0, true);
            this.lbNiveau.SetSelected(1, true);
            this.lbNiveau.SetSelected(2, true);
            this.lbNbPrefixes.Text = LogotronLib.clsConst.sHasard;
            this.lbFreq.SetSelected(0, true);
            this.lbFreq.SetSelected(1, true);
            this.lbFreq.SetSelected(2, true);
            this.lbFreq.SetSelected(3, true);
            m_bMajViaCode = false;

            clsGestBase.InitBases();

            var dicoMotsExistants = new Dictionary<string, clsMotExistant>();
            if (clsConstWinForm.sModeLectureMotsExistants == 
                enumModeLectureMotExistant.Csv)
            {
                string sChemin = Application.StartupPath + "\\MotsSimples" + clsConstWinForm.sLang + ".csv";
                clsUtilLogotronWF.ChargerMotsExistantsCsv(dicoMotsExistants, sChemin);
            }
            else if (clsConstWinForm.sModeLectureMotsExistants == 
                enumModeLectureMotExistant.Code)
            {
                clsListeMotsExistants.ChargerMotsExistantsCode(dicoMotsExistants);
            }
            
            clsGestBase.ChargerMotsExistants(dicoMotsExistants);
            //string sCheminLogotronCsv = Application.StartupPath + "\\Logotron.csv";
            //string sCheminLogotronJson = Application.StartupPath + "\\Logotron.json";
            //if (clsConstWinForm.sModeLecture == enumModeLecture.JSon)
            //    clsLogotron.LireLogotronJSon();
            //else if (clsConstWinForm.sModeLecture == enumModeLecture.Code)
                clsLogotron.LireLogotronCode();
            //clsLogotron.InitialisationPrefixes(sCheminLogotronCsv, clsConst.sModeLecture);
            //InitialisationSuffixes(sModeLecture)

            MajNbMotLogotron();
        }

        private void cmdCopier_Click(object sender, EventArgs e)
        {
            var sb = new StringBuilder();
            foreach (string sTxt in this.lbResultats.Items)
                sb.AppendLine(sTxt);
            string sTxtFinal = sb.ToString();
            string sTxtFinal2 = sTxtFinal.Replace(
                Util.clsUtil.sCarSautDeLigne + clsConstWinForm.sCrLf, "");
            if (UtilWinForm.clsMessageUtil.bCopyToClipboard(sTxtFinal2))
                MsgBox("Le texte a été copié dans le presse papier !");
        }

        private void cmdGo_Click(object sender, EventArgs e)
        {
            string sMot = "";
            string sExplication = "";
            string sDetail = "";
            List<string> lstEtym = new List<string>();
            string sNbPrefixesSuccessifs = this.lbNbPrefixes.Text;
            
            var lstNiv = new List<string>();
            //int iNumNiv = 0;
            foreach (object obj in this.lbNiveau.SelectedItems)
            {
                //iNumNiv ++;
                lstNiv.Add(obj.ToString());
            }
            //if (iNumNiv == 0) return;

            var lstFreq = new List<string>();
            foreach (object obj in this.lbFreq.SelectedItems)
            {
                string sFreqSelAbrege = obj.ToString();
                string sFreqSelComplet = enumFrequenceAbrege.sConv(sFreqSelAbrege);
                lstFreq.Add(sFreqSelComplet);
            }

            bool bGrecoLatin = this.chkOrigineGrecoLatin.Checked;
            bool bNeoRigolo = this.chkOrigineNeologisme.Checked;

            // Les préfixes et suffixes des mots du dictionnaire sont plus nombreux
            //  ne prendre que ceux qui forme des mots potentiels plausibles
            const bool bComplet = false;

            if (clsLogotron.bTirage(bComplet, sNbPrefixesSuccessifs, lstNiv, lstFreq, 
                ref sMot, ref sExplication, ref sDetail, ref lstEtym, 
                bGrecoLatin, bNeoRigolo, m_msgDelegue))
            {
                this.AfficherTexte(sMot);
                this.AfficherTexte(sExplication);
                this.AfficherTexte(sDetail);
                if (lstEtym.Count > 0)
                    foreach (string item in lstEtym) this.AfficherTexte(item);
                this.AfficherTexte("");
            }

        }

        private void MajNbMotLogotron()
        {
            List<string> lstNiv = new List<string>();
            string sNiveaux = "";
            foreach (object obj in this.lbNiveau.SelectedItems)
            {
                string sNiv = obj.ToString();
                sNiveaux += sNiv + " ";
                lstNiv.Add(sNiv);
            }
            this.AfficherTexte("Niveau(x) sélectionné(s) : " + sNiveaux);
            
            List<string> lstFreq = new List<string>();
            string sFreq = "";
            foreach (object obj in this.lbFreq.SelectedItems)
            {
                string sFreqSelAbrege = obj.ToString();
                string sFreqSelComplet = enumFrequenceAbrege.sConv(sFreqSelAbrege);
                sFreq += sFreqSelAbrege + " ";
                lstFreq.Add(sFreqSelComplet);
            }
            this.AfficherTexte("Fréquence(s) sélectionnée(s) : " + sFreq);

            bool bGrecoLatin = this.chkOrigineGrecoLatin.Checked;
            bool bNeoRigolo = this.chkOrigineNeologisme.Checked;
            int iNbPrefixes = clsGestBase.m_prefixes.iLireNbSegmentsUniques(
                lstNiv, lstFreq, bGrecoLatin, bNeoRigolo);
            int iNbSuffixes = clsGestBase.m_suffixes.iLireNbSegmentsUniques(
                lstNiv, lstFreq, bGrecoLatin, bNeoRigolo);
            long lNbPrefixesCombi = 0L;
            string sTxtCombi = "";
            string text = this.lbNbPrefixes.Text;
            if (text == "H" || text == "1")
            {
                sTxtCombi = iNbPrefixes + " préfixes";
                lNbPrefixesCombi = iNbPrefixes;
            }
            else if (text == "2")
            {
                sTxtCombi = iNbPrefixes + " préfixes x " +
                    (iNbPrefixes - 1) + " préfixes";
                lNbPrefixesCombi = iNbPrefixes * (iNbPrefixes - 1);
            }
            else if (text == "3")
            {
                sTxtCombi = iNbPrefixes + " préfixes x " +
                    (iNbPrefixes - 1) + " préfixes x " +
                    (iNbPrefixes - 2) + " préfixes";
                lNbPrefixesCombi = iNbPrefixes * (iNbPrefixes - 1) *
                    (iNbPrefixes - 2);
            }
            else if (text == "4")
            {
                sTxtCombi = iNbPrefixes + " préfixes x " +
                    (iNbPrefixes - 1) + " préfixes x " +
                    (iNbPrefixes - 2) + " préfixes x " +
                    (iNbPrefixes - 3) + " préfixes";
                lNbPrefixesCombi = (long)iNbPrefixes * (long)(iNbPrefixes - 1) * 
                    (iNbPrefixes - 2) * (iNbPrefixes - 3);
            }
            else if (text == "5")
            {
                sTxtCombi = iNbPrefixes + " préfixes x " +
                    (iNbPrefixes - 1) + " préfixes x " + 
                    (iNbPrefixes - 2) + " préfixes x " + 
                    (iNbPrefixes - 3) + " préfixes x " + 
                    (iNbPrefixes - 4) + " préfixes";
                lNbPrefixesCombi = (long)iNbPrefixes * (long)(iNbPrefixes - 1) *
                    (iNbPrefixes - 2) * (iNbPrefixes - 3) * (iNbPrefixes - 4);
            }
            long lNbCombi = lNbPrefixesCombi * iNbSuffixes;
            string sNbCombi = Util.clsUtil.sFormaterNumeriqueLong(lNbCombi);
            this.AfficherTexte(sTxtCombi + " x " + 
                (iNbSuffixes) + " suffixes = " + 
                sNbCombi + " combinaisons pour le Logotron");
            this.AfficherTexte("");

            if (this.chkOrigineGrecoLatin.Checked) return;
            var lstPrefixes = clsGestBase.m_prefixes.lstSegmentsAutreOrigine(lstNiv, bNeoRigolo);
            foreach (clsSegmentBase prefixe in lstPrefixes)
                AfficherTexte(prefixe.sAfficher(bPrefixe: true));
            var lstSuffixes = clsGestBase.m_suffixes.lstSegmentsAutreOrigine(lstNiv, bNeoRigolo);
            foreach (clsSegmentBase suffixe in lstSuffixes)
                AfficherTexte(suffixe.sAfficher(bPrefixe: false));
            this.AfficherTexte("");
        }

        private void lbNiveau_SelectedValueChanged(object sender, EventArgs e)
        {
            if (m_bMajViaCode) return;
            MajNbMotLogotron();
        }
        
        private void lbFreq_SelectedValueChanged(object sender, EventArgs e)
        {
            if (m_bMajViaCode) return;
            MajNbMotLogotron();
        }

        private void lbNbPrefixes_SelectedValueChanged(object sender, EventArgs e)
        {
            if (m_bMajViaCode) return;
            MajNbMotLogotron();
        }
        
        private void chkOrigineGrecoLatin_Click(object sender, EventArgs e)
        {
            if (m_bMajViaCode) return;
            if (this.chkOrigineGrecoLatin.Checked) this.chkOrigineNeologisme.Checked = false;
            MajNbMotLogotron();
        }
        
        private void chkOrigineNeoRigolo_Click(object sender, EventArgs e)
        {
            if (m_bMajViaCode) return;
            if (this.chkOrigineNeologisme.Checked) this.chkOrigineGrecoLatin.Checked = false;
            MajNbMotLogotron();
        }

        private void cmdQuiz_Click(object sender, EventArgs e)
        {
            using (Logotron.Src.frmQuiz frmQuiz0 = new Logotron.Src.frmQuiz())
            {
                frmQuiz0.StartPosition = FormStartPosition.CenterScreen;
                frmQuiz0.ShowDialog(this);
            }
        }
    }
}

File: Src/frmLogotron.vb

' Logotron : jouer avec les préfixes et les suffixes de la langue française
' -------------------------------------------------------------------------
' https://github.com/PatriceDargenton/Logotron
' Documentation : Logotron.html
' http://patrice.dargenton.free.fr/CodesSources/Logotron/index.html
' http://patrice.dargenton.free.fr/CodesSources/Logotron/Logotron.vbproj.html
' Version 1.04 du 05/05/2018 : Gestion des élisions (ex.: palé(o) - onto - logie)
' Version 1.01 du 02/09/2018 : Première version
' Par Patrice Dargenton : mailto:patrice.dargenton@free.fr
' http://patrice.dargenton.free.fr/index.html
' http://patrice.dargenton.free.fr/CodesSources/index.html
' -------------------------------------------------------------------------

' D'après la source :
' https://www.jp-petit.org/Divers/LOGOTRON/logotron.HTM

' Conventions de nommage des variables :
' ------------------------------------
' b pour Boolean (booléen vrai ou faux)
' i pour Integer : % (en VB .Net, l'entier a la capacité du VB6.Long)
' l pour Long : &
' r pour nombre Réel (Single!, Double# ou Decimal : D)
' s pour String : $
' c pour Char ou Byte
' d pour Date
' u pour Unsigned (non signé : entier positif)
' a pour Array (tableau) : ()
' m_ pour variable Membre de la classe ou de la feuille (Form)
'  (mais pas pour les constantes)
' frm pour Form
' cls pour Classe
' mod pour Module
' ...
' ------------------------------------

Imports System.Text

Public Class frmLogotron

    Private m_bMajViaCode As Boolean = False

    Private WithEvents m_msgDelegue As clsMsgDelegue = New clsMsgDelegue

    Private Sub Form1_Load(sender As Object, e As EventArgs) Handles Me.Load

        Dim sVersion$ = " - V" & sVersionAppli & " (" & sDateVersionAppli & ")"
        Dim sDebug$ = " - Debug"
        Dim sTxt$ = Me.Text & sVersion
        If bDebug Then sTxt &= sDebug
        Me.Text = sTxt

        Initialisation(bAfficherAvert:=False)

        m_bMajViaCode = True
        Me.lbNiveau.SetSelected(0, True)
        Me.lbNiveau.SetSelected(1, True)
        Me.lbNiveau.SetSelected(2, True)
        Me.lbNbPrefixes.Text = sHasard
        Me.lbFreq.SetSelected(0, True)
        Me.lbFreq.SetSelected(1, True)
        Me.lbFreq.SetSelected(2, True)
        Me.lbFreq.SetSelected(3, True)
        m_bMajViaCode = False

        MajNbMotsLogotron()

    End Sub

    Private Sub Initialisation(bAfficherAvert As Boolean)

        EffacerMessages()
        InitBases()
        If sModeLectureMotsExistants = enumModeLectureMotExistant.sCsv Then
            ChargerMotsExistantsCsv()
        ElseIf sModeLectureMotsExistants = enumModeLectureMotExistant.sCode Then
            ChargerMotsExistantsCode(m_dicoMotsExistants)
        End If

        Dim sCheminLogotronCsv$ = Application.StartupPath & "\Logotron" & sLang & ".csv"
        Dim sCheminSensConcept$ = Application.StartupPath & "\SensConcept" & sLang & ".csv"
        InitialisationPrefixes(sCheminLogotronCsv, sModeLecture, m_msgDelegue)
        InitialisationSuffixes(sModeLecture)
        TraiterEtExporterDonnees(bAfficherAvert, m_msgDelegue, sCheminSensConcept)

    End Sub

    Private Sub cmdAvert_Click(sender As Object, e As EventArgs) Handles cmdAvert.Click
        Initialisation(bAfficherAvert:=True)
    End Sub

    Private Sub cmdAnnuler_Click(sender As Object, e As EventArgs) Handles cmdAnnuler.Click
        m_msgDelegue.m_bAnnuler = True
    End Sub

    Private Sub AfficherMessage(sender As Object, e As clsMsgEventArgs) _
        Handles m_msgDelegue.EvAfficherMessage
        AfficherTexte(e.sMessage)
    End Sub

    Private m_iIndex% = 0
    Private Sub AfficherTexte(sTxt$)
        AfficherTexteListBox(sTxt, m_iIndex, Me, Me.lbResultats)
    End Sub
    Private Sub EffacerMessages()
        lbResultats.Items.Clear()
        m_iIndex = 0
    End Sub

    Private Sub lbNiveau_SelectedValueChanged(sender As Object, e As EventArgs) _
        Handles lbNiveau.SelectedValueChanged

        If m_bMajViaCode Then Exit Sub
        MajNbMotsLogotron()

    End Sub

    Private Sub lbFreq_SelectedValueChanged(sender As Object, e As EventArgs) _
        Handles lbFreq.SelectedValueChanged

        If m_bMajViaCode Then Exit Sub
        MajNbMotsLogotron()

    End Sub

    Private Sub lbNbPrefixes_SelectedValueChanged(sender As Object, e As EventArgs) _
        Handles lbNbPrefixes.SelectedValueChanged

        If m_bMajViaCode Then Exit Sub
        MajNbMotsLogotron()

    End Sub

    Private Sub chkOrigineGrecoLatin_Click(sender As Object, e As EventArgs) _
        Handles chkOrigineGrecoLatin.Click

        If m_bMajViaCode Then Exit Sub
        If Me.chkOrigineGrecoLatin.Checked Then Me.chkOrigineNeoRigolo.Checked = False
        MajNbMotsLogotron()

    End Sub

    Private Sub chkOrigineNeoRigolo_Click(sender As Object, e As EventArgs) _
        Handles chkOrigineNeoRigolo.Click

        If m_bMajViaCode Then Exit Sub
        If Me.chkOrigineNeoRigolo.Checked Then Me.chkOrigineGrecoLatin.Checked = False
        MajNbMotsLogotron()

    End Sub

    Private Sub Sablier(Optional bDesactiver As Boolean = False)

        ' Me.Cursor : Curseur de la fenêtre
        ' Cursor.Current : Curseur de l'application

        If bDesactiver Then
            Me.Cursor = Cursors.Default
        Else
            Me.Cursor = Cursors.WaitCursor
        End If

    End Sub

    Private Sub Activation(bActiver As Boolean)

        m_msgDelegue.m_bAnnuler = False

        Me.cmdAnnuler.Enabled = Not bActiver
        Sablier(bDesactiver:=bActiver)

        Me.cmdCopier.Enabled = bActiver
        Me.cmdGo.Enabled = bActiver
        Me.cmdQuiz.Enabled = bActiver
        Me.cmdAvert.Enabled = bActiver

    End Sub

    Private Sub MajNbMotsLogotron()

        Dim lstNiv As New List(Of String)
        Dim sNiveaux$ = ""
        For Each obj In Me.lbNiveau.SelectedItems
            sNiveaux &= obj.ToString & " "
            lstNiv.Add(obj.ToString)
        Next
        AfficherTexte("Niveau(x) sélectionné(s) : " & sNiveaux)

        Dim lstFreq As New List(Of String)
        Dim sFreq$ = ""
        For Each obj In Me.lbFreq.SelectedItems
            Dim sFreqSelAbrege$ = obj.ToString
            Dim sFreqSelComplet$ = enumFrequenceAbrege.sConv(sFreqSelAbrege)
            sFreq &= sFreqSelAbrege & " "
            lstFreq.Add(sFreqSelComplet)
        Next
        AfficherTexte("Fréquence(s) sélectionnée(s) : " & sFreq)

        Dim bGrecoLatin = Me.chkOrigineGrecoLatin.Checked
        Dim bNeoRigolo = Me.chkOrigineNeoRigolo.Checked
        Dim iNbPrefixes = m_prefixes.iLireNbSegmentsUniques(lstNiv, lstFreq, bGrecoLatin, bNeoRigolo)
        Dim iNbSuffixes = m_suffixes.iLireNbSegmentsUniques(lstNiv, lstFreq, bGrecoLatin, bNeoRigolo)
        Dim lNbPrefixesCombi& = 0
        Dim sTxtCombi$ = ""

        Select Case Me.lbNbPrefixes.Text
            Case "H", "1"
                sTxtCombi =
                iNbPrefixes & " préfixes"
                lNbPrefixesCombi = iNbPrefixes
            Case "2"
                sTxtCombi =
                iNbPrefixes & " préfixes x " &
                (iNbPrefixes - 1) & " préfixes"
                lNbPrefixesCombi = iNbPrefixes * (iNbPrefixes - 1)
            Case "3"
                sTxtCombi =
                iNbPrefixes & " préfixes x " &
                (iNbPrefixes - 1) & " préfixes x " &
                (iNbPrefixes - 2) & " préfixes"
                lNbPrefixesCombi = iNbPrefixes * (iNbPrefixes - 1) * (iNbPrefixes - 2)
            Case "4"
                sTxtCombi =
                iNbPrefixes & " préfixes x " &
                (iNbPrefixes - 1) & " préfixes x " &
                (iNbPrefixes - 2) & " préfixes x " &
                (iNbPrefixes - 3) & " préfixes"
                lNbPrefixesCombi = CLng(iNbPrefixes) * (iNbPrefixes - 1) * (iNbPrefixes - 2) *
                (iNbPrefixes - 3)
            Case "5"
                sTxtCombi =
                iNbPrefixes & " préfixes x " &
                (iNbPrefixes - 1) & " préfixes x " &
                (iNbPrefixes - 2) & " préfixes x " &
                (iNbPrefixes - 3) & " préfixes x " &
                (iNbPrefixes - 4) & " préfixes"
                lNbPrefixesCombi = CLng(iNbPrefixes) * (iNbPrefixes - 1) * (iNbPrefixes - 2) *
                (iNbPrefixes - 3) * (iNbPrefixes - 4)
        End Select
        Dim lNbCombi& = lNbPrefixesCombi * iNbSuffixes
        Dim sNbCombi = sFormaterNumeriqueLong(lNbCombi)

        AfficherTexte(sTxtCombi & " x " & iNbSuffixes & " suffixes = " &
            sNbCombi & " combinaisons pour le Logotron")
        AfficherTexte("")

        If Not Me.chkOrigineGrecoLatin.Checked Then
            Dim lstPrefixes = m_prefixes.lstSegmentsAutreOrigine(lstNiv, lstFreq, bNeoRigolo)
            For Each prefixe In lstPrefixes
                AfficherTexte(prefixe.sAfficher(bPrefixe:=True))
            Next
            Dim lstSuffixes = m_suffixes.lstSegmentsAutreOrigine(lstNiv, lstFreq, bNeoRigolo)
            For Each suffixe In lstSuffixes
                AfficherTexte(suffixe.sAfficher(bPrefixe:=False))
            Next
        End If

    End Sub

    Private Sub cmdCopier_Click(sender As Object, e As EventArgs) Handles cmdCopier.Click

        Dim sTxt$ = sLireListBox(lbResultats)
        If bCopierPressePapier(sTxt) Then _
            MsgBox(sMsgCopiePressePapier, MsgBoxStyle.Exclamation, m_sTitreMsg)

    End Sub

    Private Sub cmdGo_Click(sender As Object, e As EventArgs) Handles cmdGo.Click

        Dim sMot$ = "", sExplication$ = "", sDetail$ = ""
        Dim lstEtym As New List(Of String)

        ' Les préfixes et suffixes des mots du dictionnaire sont plus nombreux
        '  ne prendre que ceux qui forme des mots potentiels plausibles
        Const bComplet As Boolean = False

        Dim sNbPrefixesSuccessifs$ = Me.lbNbPrefixes.Text

        Dim lstNiv As New List(Of String)
        Dim iNumNiv% = 0
        For Each obj In Me.lbNiveau.SelectedItems
            iNumNiv += 1
            lstNiv.Add(obj.ToString)
        Next
        If iNumNiv = 0 Then Exit Sub

        Dim lstFreq As New List(Of String)
        Dim iNumFreq% = 0
        For Each obj In Me.lbFreq.SelectedItems
            iNumFreq += 1
            Dim sFreqSelAbrege$ = obj.ToString
            Dim sFreqSelComplet$ = enumFrequenceAbrege.sConv(sFreqSelAbrege)
            lstFreq.Add(sFreqSelComplet)
        Next
        If iNumFreq = 0 Then Exit Sub

        Dim bGrecoLatin = Me.chkOrigineGrecoLatin.Checked
        Dim bNeoRigolo = Me.chkOrigineNeoRigolo.Checked

        If Not bTirage(bComplet, sNbPrefixesSuccessifs, lstNiv, lstFreq, bGrecoLatin,
            bNeoRigolo, sMot, sExplication, sDetail, lstEtym) Then Exit Sub

        AfficherTexte(sMot)
        AfficherTexte(sExplication)
        AfficherTexte(sDetail)
        If lstEtym.Count > 0 Then
            For Each sEtym In lstEtym
                AfficherTexte(sEtym)
            Next
        End If
        AfficherTexte("")

    End Sub

    Private Sub cmdQuiz_Click(sender As Object, e As EventArgs) Handles cmdQuiz.Click

        Activation(bActiver:=False)

        Using frmQuiz0 As New frmQuiz
            frmQuiz0.StartPosition = FormStartPosition.CenterScreen
            frmQuiz0.ShowDialog(Me)
        End Using

        Activation(bActiver:=True)

    End Sub

End Class
